src/main/java/dev/hephaestus/glowcase/block/entity/HyperlinkBlockEntity.java

```diff
@@ -13,48 +14,52 @@
 import org.jetbrains.annotations.Nullable;
 
 public class HyperlinkBlockEntity extends BlockEntity {
-	private String url = "";
-
-	public HyperlinkBlockEntity(BlockPos pos, BlockState state) {
-		super(Glowcase.HYPERLINK_BLOCK_ENTITY, pos, state);
-	}
-
-	public String getUrl() {
-		return url;
-	}
-
-	public void setUrl(String newUrl) {
-		url = newUrl;
-		markDirty();
-		dispatch();
-	}
-
-	@Override
-	public void writeNbt(NbtCompound tag, RegistryWrapper.WrapperLookup registryLookup) {
-		super.writeNbt(tag, registryLookup);
-		tag.putString("url", this.url);
-	}
-
-	@Override
-	public void readNbt(NbtCompound tag, RegistryWrapper.WrapperLookup registryLookup) {
-		super.readNbt(tag, registryLookup);
-		this.url = tag.getString("url");
-	}
-
-	// standard blockentity boilerplate
-
-	public void dispatch() {
-		if (world instanceof ServerWorld sworld) sworld.getChunkManager().markForUpdate(pos);
-	}
-
-	@Override
-	public NbtCompound toInitialChunkDataNbt(RegistryWrapper.WrapperLookup registryLookup) {
-		return createNbt(registryLookup);
-	}
-
-	@Nullable
-	@Override
-	public Packet<ClientPlayPacketListener> toUpdatePacket() {
-		return BlockEntityUpdateS2CPacket.create(this);
-	}
+    private String url = "";
+
+    public HyperlinkBlockEntity(BlockPos pos, BlockState state) {
+        super(Glowcase.HYPERLINK_BLOCK_ENTITY, pos, state);
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String newUrl) {
+        url = newUrl;
+        markDirty();
+        dispatch();
+    }
+
+    @Override
+    public void writeNbt(NbtCompound tag, RegistryWrapper.WrapperLookup registryLookup) {
+        super.writeNbt(tag, registryLookup);
+        tag.putString("url", this.url);
+    }
+
+    @Override
+    public void readNbt(NbtCompound tag, RegistryWrapper.WrapperLookup registryLookup) {
+        super.readNbt(tag, registryLookup);
+        this.url = tag.getString("url");
+    }
+
+    public EditHyperlinkBlock createPacketData() {
```

Comment:
I might rename this to "createEditPacket" since I think that's more descriptive. But that'd be in a separate codestyle pass, nothing that would block this pr

Reply:
I guess this would apply for all the other BEs too, then?

I wonder if something like an interface or abstract super class would be good to enforce / abstract this whole structure a bit better.

Reply:
Also, I think I will move from short to byte for the enum packet codec entries too.

If there is ever going to be an enum with more entries than the byte size, there is something definitely wrong with the enum, and it should probably refactored to be something like a list or map...

Reply:
my suggestion on this - `EditPacket<T extends BlockEntity> extends CustomPayload`

contains
- `void send()` - defaulted c2s send.
- `void receive(context)` - abstract for packets to implement
- `EditPacket create(T blockEntity)` - construct packet from blockentity.

Anything that's not public from the BE that needs to be, just make it public.